src/test/java/io/aiven/kafka/connect/opensearch/BulkProcessorTest.java

```diff
@@ -486,13 +494,140 @@ public void ignoreOnVersionConfict(final @Mock RestHighLevelClient client) throw
 
         bulkProcessor.start();
 
-        bulkProcessor.add(newIndexRequest(42), 1);
-        bulkProcessor.add(newIndexRequest(43), 1);
+        bulkProcessor.add(newIndexRequest(42), newSinkRecord(), 1);
+        bulkProcessor.add(newIndexRequest(43), newSinkRecord(), 1);
         bulkProcessor.flush(1000);
 
         assertTrue(clientAnswer.expectationsMet());
     }
 
+    @Test
+    public void reportToDlqWhenVersionConflictBehaviorIsReport(final @Mock RestHighLevelClient client)
+            throws IOException {
+        final var clientAnswer = new ClientAnswer();
+        when(client.bulk(any(BulkRequest.class), eq(RequestOptions.DEFAULT))).thenAnswer(clientAnswer);
+
+        final var dlqReporter = mock(ErrantRecordReporter.class);
+        final var config = new OpensearchSinkConnectorConfig(Map.of(
+                CONNECTION_URL_CONFIG, "http://localhost",
+                MAX_BUFFERED_RECORDS_CONFIG, "100",
+                MAX_IN_FLIGHT_REQUESTS_CONFIG, "5",
+                BATCH_SIZE_CONFIG, "2",
+                LINGER_MS_CONFIG, "1000",
+                MAX_RETRIES_CONFIG, "3",
+                READ_TIMEOUT_MS_CONFIG, "1",
+                DLQ_TOPIC_NAME_CONFIG, "dlq_topic",
```

Comment:
This line can be removed and the tests will still pass. IMO it's better to remove it since it better illustrates that the only thing that the `BulkProcessor` cares about is whether it's configured with the `report` value for handling version conflicts or malformed docs, and doesn't care about framework-level properties like the DLQ topic name or errors tolerance setting.